src/battle_ai_main.c

```diff
@@ -4173,6 +4173,11 @@ static u32 AI_CalcMoveScore(u32 battlerAtk, u32 battlerDef, u32 move)
         IncreaseStatUpScore(battlerAtk, battlerDef, STAT_CHANGE_SPATK, &score);
         IncreaseStatUpScore(battlerAtk, battlerDef, STAT_CHANGE_ATK, &score);
         break;
+    case EFFECT_TIDY_UP:
```

Comment:
I would recommend using a personal scoring system for Tidy Up. The AI should determine the pros and cons of using Tidy Up and act accordingly. Such as prefer less to remove it or its partner's substitute, but prefer more to remove hazards on its side or boost stats. Might be worth speaking to Showdown players to see how they'd determine when it would be worth using the move.

Reply:
Finally added a function. 

I think a lot of the decisions I've made are very subjective but basically I've decided that hazards on users field are more important to remove. Removing a Substitute on the opposing field should also be preferred. In contrast if the user has a Substitute and is faster then opposing mon the score will be decreased by -10.

Reply:
Overall looks fine. Definitely warrants battle testing though because realistically the AI is going to be making mistakes with this (but that's normal for AI logic).